### PR TITLE
Devel

### LINKED ISSUE #3: talos_data has a dependency to Gazebo set to required plus informations to be used outside the gazebo use_case
The dependency is due to the gazebo plugin suppose to simulate the spring in the hip.
This has been set to required in
 https://github.com/stack-of-tasks/talos-data/blob/5514692d4b81c22c61ad82a995cf267bf058929a/CMakeLists.txt#L26
Unfortunately they are additional information specifically for the walking pattern generator which are set here:
https://github.com/stack-of-tasks/talos-data/blob/master/srdf/talos_wpg.srdf

Due to this jrl-walkgen has a dependency to talos-data and then to gazebo. The later is not welcome when using jrl-walkgen. The problem is that right now it is difficult programmatically to cut the dependency to gazebo. 

They are several other solutions:

- [ ] Move the data related to the wpg in a different place (I would prefer to avoid another package) example-robot-data should stay clean IMHO
- [ ] Move the gazebo plugin to another package talos_data_gazebo (for the same reason I do not think this is worth it)
- [ ] Make Gazebo optional when needed at compilation, and installed system-wide for the binary distribution.

I feel that the third solution is better for the long term.

Happy to discuss simple and fast solution.

### PR DESCRIPTION
**Pull request by olivier-stasse**
Modify CMakeLists.txt to have Gazebo as an optional dependency. Fix #3

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 3.1)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.17)
```
Comment (nim65s) [inline CMakeLists.txt:1]: I don't think that was intentional

**Review by nim65s — COMMENTED** [review]
Hi,

I still don't see the point of making Gazebo optional. Could you provide a scenario where this is useful ?

**Comment by olivier-stasse** [comment]
talos_data can be used without ros, because the urdf files have been generated.
Gazebo required implies to have this dependency installed.
talos_data can be used with jrl-walkgen without the SoT and I am working on examples of the SoT without ROS or Gazebo.
The best solution is to remove the Spring plugin.  We could have a gepetto-gazebo-pkgs (where we could put the ouster simulator).

**Comment by nim65s** [comment]
superseeded by #8

**Pull request closed without merging**